includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -776,19 +777,24 @@ protected function getArticlesDetails( $articleIds, $articleKeys = [], $width =
 		$collection = $this->appendMetadata( $collection );
 
 		$thumbnails = null;
-		//if strict return to original ids order
+		$collection = $this->preserveOriginalOrder( $articleIds, $collection );
```

Comment:
What's the rationale behind moving the behaviour responsible for preserving the order from inside of the loop to outside of it? (all vs strict = true)